src/twisted/words/test/test_irc.py

```diff
@@ -792,14 +806,14 @@ def test_support_CHANMODES(self):
         self._testFeatureDefault('CHANMODES', [('CHANMODES', 'b,,lk,ha,ha')])
 
         self.assertEqual(
-            self._parseFeature('CHANMODES', ''),
+            self._parseFeature('CHANMODES', ',,,'),
```

Comment:
This is needed.  Without this change, the test will fail on Python 3 like this:

```
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/Users/crodrigues/twisted13/src/twisted/words/test/test_irc.py", line 821, in test_support_CHANMODES
    'noParam': ''})
  File "/Users/crodrigues/twisted13/src/twisted/trial/_synctest.py", line 425, in assertEqual
    super(_Assertions, self).assertEqual(first, second, msg)
  File "/usr/local/Cellar/python3/HEAD-42a10253d066_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", line 820, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/usr/local/Cellar/python3/HEAD-42a10253d066_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", line 1111, in assertDictEqual
    self.fail(self._formatMessage(msg, standardMsg))
twisted.trial.unittest.FailTest: {'addressModes': ''} != {'param': '', 'noParam': '', 'addressModes': '', 'setParam': ''}
- {'addressModes': ''}
+ {'addressModes': '', 'noParam': '', 'param': '', 'setParam': ''}

twisted.words.test.test_irc.ServerSupportedFeatureTests.test_support_CHANMODES
-------------------------------------------------------------------------------
```

The underlying _parseChanModesParam() function is using a combination of map, lambda, and dict which is triggering this difference in behavior between 2 and 3.